### PR TITLE
Implement `Position.compare/2`

### PR DESCRIPTION
**Pull request by zachallaun**
Requires #763

https://github.com/zachallaun/lexical/compare/za-exclusive-range-contains...zachallaun:lexical:za-position-compare?expand=1

### CODE REVIEW

File: projects/lexical_shared/lib/lexical/document/position.ex [hunk 1]
```diff
@@ -70,6 +70,33 @@ defmodule Lexical.Document.Position do
...
+
+  Returns `:gt`, `:lt`, or `:eq` depending on the location of the first
+  position relative to the second.
+  """
+  @spec compare(t | {line, character}, t | {line, character}) :: :lt | :eq | :gt
+  def compare(%__MODULE__{} = pos1, %__MODULE__{} = pos2) do
+    compare({pos1.line, pos1.character}, {pos2.line, pos2.character})
+  end
+
+  def compare(%__MODULE__{} = pos1, pos2) do
```
Comment (scohen) [inline projects/lexical_shared/lib/lexical/document/position.ex:85]:
```suggestion
  def compare(%__MODULE__{} = pos1, {_, _} = pos2) do
```

File: projects/lexical_shared/lib/lexical/document/position.ex [hunk 2]
```diff
@@ -70,6 +70,33 @@ defmodule Lexical.Document.Position do
...
+    compare({pos1.line, pos1.character}, pos2)
+  end
+
+  def compare(pos1, %__MODULE__{} = pos2) do
```
Comment (scohen) [inline projects/lexical_shared/lib/lexical/document/position.ex:89]:
```suggestion
  def compare({_, _} = pos1, %__MODULE__{} = pos2) do
```

File: projects/lexical_shared/lib/lexical/document/position.ex [hunk 3]
```diff
@@ -70,6 +70,33 @@ defmodule Lexical.Document.Position do
...
+    compare(pos1, {pos2.line, pos2.character})
+  end
+
+  def compare({_, _} = first, {_, _} = second) do
```
Comment (scohen) [inline projects/lexical_shared/lib/lexical/document/position.ex:93]: this feels a little dangerous, should we constrain line and character to be integers?

**Review by scohen — COMMENTED** [review]
The changes in the cursors worry me a little bit, can we talk on discord?

File: projects/lexical_shared/lib/lexical/document/position.ex (hunk 3 above)
Comment (zachallaun) [reply projects/lexical_shared/lib/lexical/document/position.ex:93]: Yeah, you're right, especially considering it's ultimately doing term comparison. I'll add guards.

**Comment by zachallaun** [comment]
Closing in favor of #767

**Pull request closed without merging**